crates/uv/src/commands/build.rs

```diff
@@ -0,0 +1,338 @@
+
+/// Build source distributions and wheels.
+#[allow(clippy::fn_params_excessive_bools)]
+pub(crate) async fn build(
+    path: Option<PathBuf>,
+    sdist: bool,
+    wheel: bool,
+    python: Option<String>,
+    settings: ResolverSettings,
+    no_config: bool,
+    python_preference: PythonPreference,
+    python_downloads: PythonDownloads,
+    connectivity: Connectivity,
+    concurrency: Concurrency,
+    native_tls: bool,
+    cache: &Cache,
+    printer: Printer,
+) -> Result<ExitStatus> {
+    let assets = build_impl(
+        path.as_deref(),
+        sdist,
+        wheel,
+        python.as_deref(),
+        settings.as_ref(),
+        no_config,
+        python_preference,
+        python_downloads,
+        connectivity,
+        concurrency,
+        native_tls,
+        cache,
+        printer,
+    )
+    .await?;
+
+    match assets {
+        BuiltDistributions::Wheel(wheel) => {
+            anstream::eprintln!("Successfully built {}", wheel.bold().cyan());
+        }
+        BuiltDistributions::Sdist(sdist) => {
+            anstream::eprintln!("Successfully built {}", sdist.bold().cyan());
+        }
+        BuiltDistributions::Both(sdist, wheel) => {
+            anstream::eprintln!(
+                "Successfully built {} and {}",
+                sdist.bold().cyan(),
+                wheel.bold().cyan()
+            );
+        }
+    }
+
+    Ok(ExitStatus::Success)
+}
+
+#[allow(clippy::fn_params_excessive_bools)]
+async fn build_impl(
+    path: Option<&Path>,
+    sdist: bool,
+    wheel: bool,
+    python_request: Option<&str>,
+    settings: ResolverSettingsRef<'_>,
+    no_config: bool,
+    python_preference: PythonPreference,
+    python_downloads: PythonDownloads,
+    connectivity: Connectivity,
+    concurrency: Concurrency,
+    native_tls: bool,
+    cache: &Cache,
+    printer: Printer,
+) -> Result<BuiltDistributions> {
+    // Extract the resolver settings.
+    let ResolverSettingsRef {
+        index_locations,
+        index_strategy,
+        keyring_provider,
+        allow_insecure_host,
+        resolution: _,
+        prerelease: _,
+        config_setting,
+        no_build_isolation,
+        no_build_isolation_package,
+        exclude_newer,
+        link_mode,
+        upgrade: _,
+        build_options,
+        sources,
+    } = settings;
+
+    let client_builder = BaseClientBuilder::default()
+        .connectivity(connectivity)
+        .native_tls(native_tls);
+
+    // (1) Explicit request from user
+    let mut interpreter_request = python_request.map(PythonRequest::parse);
+
+    // (2) Request from `.python-version`
+    if interpreter_request.is_none() {
+        interpreter_request = PythonVersionFile::discover(&*CWD, no_config, false)
+            .await?
+            .and_then(PythonVersionFile::into_version);
+    }
+
+    // (3) `Requires-Python` in `pyproject.toml`
+    if interpreter_request.is_none() {
```

Comment:
Can we support `--no-project` here? Trying to do that everywhere. Might not make sense for this command though?